multicore_runtime/dispatcher.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A WSGI app that, once configured, dispatches requests to user apps."""
+
+import logging
+import re
+
+from werkzeug.wrappers import Request
```

Comment:
Since I learned Python outside of Google, I'm used to it and use Ctrl-F or functionality in my editor habitually to check defined names. But I don't think using the full name definition is worse, it's just different, and don't mind fixing it for in-package stuff if that's the decision.

As for out-of-package... I think we should adhere to the standard set by Werkzeug in its own examples if possible, but I'm flexible on that point. Thanks for starting a thread on it, let me know what people say.